akka-stream/src/main/scala/akka/stream/impl/SplitAfterProcessorImpl.scala

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (C) 2009-2014 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.impl
+
+import scala.util.control.NonFatal
+import akka.actor.Props
+import akka.stream.ActorFlowMaterializerSettings
+import akka.stream.Supervision
+import akka.stream.scaladsl.Source
+
+/**
+ * INTERNAL API
+ */
+private[akka] object SplitAfterProcessorImpl {
+  def props(settings: ActorFlowMaterializerSettings, splitPredicate: Any ⇒ Boolean): Props =
+    Props(new SplitAfterProcessorImpl(settings, splitPredicate))
+
+  private trait SplitDecision
```

Comment:
Cool, last question then - with the new class we're able to expose: `def split(Out => SplitDecision)` should we then expose `splitWhen`, `splitAfter` and `split` (split can be used fro dropping elements at will and SplitDrop as well).

2 new methods API wise, but seems pretty powerful, sounds good?